core/src/main/java/org/kiwix/kiwixmobile/core/extensions/UriExtensions.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.extensions
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import androidx.core.content.ContextCompat.startActivity
+import org.kiwix.kiwixmobile.core.utils.AlertDialogShower
+import org.kiwix.kiwixmobile.core.utils.EXTRA_EXTERNAL_LINK
+import org.kiwix.kiwixmobile.core.utils.KiwixDialog
+import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil
+
+fun Uri.browserIntent(
```

Comment:
I more of meant 
```
Uri.browserIntent()=  Intent(Intent.ACTION_VIEW, this)
```

Hopefully a review of `EXTRA_EXTERNAL_LINK` so we can get rid of it

And I think more attention needs to be paid to the original function, I think it was doing more?